agimus_controller/agimus_controller/warm_start_reference.py

```diff
@@ -0,0 +1,122 @@
+
+    Methods:
+        setup(rmodel: pin.Model) -> None:
+            Initializes the robot model and its associated data structure for later computations.
+
+        generate(
+            initial_state: TrajectoryPoint,
+            reference_trajectory: list[TrajectoryPoint],
+        ) -> tuple[
+            npt.NDArray[np.float64],
+            list[npt.NDArray[np.float64]],
+            list[npt.NDArray[np.float64]],
+        ]:
+            Generates the initial state, reference state vectors, and control inputs for warmstart.
+
+            Parameters:
+                initial_state (TrajectoryPoint): The starting state of the robot, containing joint configuration
+                    and velocity information.
+                reference_trajectory (list[TrajectoryPoint]): A sequence of desired trajectory points, each containing
+                    joint configuration, velocity, and acceleration.
+
+            Returns:
+                tuple: A tuple containing:
+                    - x0 (npt.NDArray[np.float64]): The initial state vector `[q, v]` where `q` is the joint configuration
+                    and `v` is the joint velocity.
+                    - x_init (list[npt.NDArray[np.float64]]): A list of state vectors `[q, v]` constructed from the
+                    reference trajectory.
+                    - u_init (list[npt.NDArray[np.float64]]): A list of control inputs computed using inverse dynamics
+                    (RNEA) based on the reference trajectory.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._rmodel: pin.Model | None = None
+        self._rdata: pin.Data | None = None
+
+    def setup(self, rmodel: pin.Model) -> None:
+        self._rmodel = rmodel
+        self._rdata = self._rmodel.createData()
+
+    @override
+    def generate(
+        self,
+        initial_state: TrajectoryPoint,
+        reference_trajectory: list[TrajectoryPoint],
+    ) -> tuple[
+        npt.NDArray[np.float64],
+        list[npt.NDArray[np.float64]],
+        list[npt.NDArray[np.float64]],
+    ]:
+        """
+        Generate initial values for a warm-start of the optimization problem.
+        The state vector is `[q, v]`, where:
+            - `q` is the robot's joint configuration.
+            - `v` is the robot's joint velocity.
+        - `init_xs`: A list of state vectors `[q, v]` constructed from the reference trajectory.
+        - `init_us`: A list of control inputs computed using inverse dynamics (RNEA)
+            based on the reference trajectory.
+        """
+        # Ensure the robot model (_rmodel) is initialized before proceeding
+        assert (
+            self._rmodel is not None
+        ), "Robot model is missing in warmstart. please use warmstart.setup(rmodel)"
+
+        x0 = np.concatenate(
+            [initial_state.robot_configuration, initial_state.robot_velocity]
+        )
+        assert x0.shape[0] == (self._rmodel.nq + self._rmodel.nv), (
```

Comment:
Couldn't `self._rmodel.nq + self._rmodel.nv` be simply stored in a variable like `state_size`? Maybe even `self._state_size`, that would be updated in `setup()` function